### PR TITLE
E2E tests: separate tests linux and windows (Brute Force, FIM, Vulnerability Detector)

### PR DESCRIPTION
**Pull request by BelenValdivia**
|Related issue|
|-------------|
|     https://github.com/wazuh/wazuh-qa/issues/3156        |

## Description
Separated tests linux and windows

### Updated

- Brute Force tests
- FIM tests
- Vulnerability Detector tests

---

### Brute Force:
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9302831/R1-tests-brute-force.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9302833/R2-tests-brute-force.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9302836/R3-tests-brute-force.zip) |    Windows and Linux     |   169c9200752fc07a4d07ccc854ec7d59179d7ac2     | Nothing to highlight |
| @juliamagan  (Reviewer)   |     test_brute_force/      | :no_entry_sign: :no_entry_sign: :no_entry_sign: |  [🟢 ](https://github.com/wazuh/wazuh-qa/files/9357936/R1-3164-test-brute-force.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9357940/R2-3164-test-brute-force.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9357946/R3-3164-test-brute-force.zip) |    CentOS and Windows    |     76764fc    | Nothing to highlight |

### FIM:
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9302839/R1-tests-fim.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9302840/R2-tests-fim.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9302841/R3-tests-fim.zip)  |  Windows and Linux    |    169c9200752fc07a4d07ccc854ec7d59179d7ac2     | Nothing to highlight |
| @juliamagan  (Reviewer)   |     test_fim/      | :no_entry_sign: :no_entry_sign: :no_entry_sign: |  [🟢 ](https://github.com/wazuh/wazuh-qa/files/9358150/R1-3164-test-fim.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9358157/R2-3164-test-fim.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9358159/R3-3164-test-fim.zip) |    CentOS and Windows    |     76764fc    | Nothing to highlight |

### Vulnerability Detector:
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9302842/R1-tests-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9302843/R2-tests-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9302844/R3-tests-vulnerability-detector.zip) |    Windows and Linux     |    169c9200752fc07a4d07ccc854ec7d59179d7ac2    | Nothing to highlight |
| @juliamagan  (Reviewer)   |     test_fim/      | :no_entry_sign: :no_entry_sign: :no_entry_sign: | [🟢 ](https://github.com/wazuh/wazuh-qa/files/9358429/R1-3164-test-vulnerability-detector.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9358431/R2-3164-test-vulnerability-detector.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9358434/R3-3164-test-vulnerability-detector.zip) |    CentOS, Ubuntu and Windows    |     76764fc    | Nothing to highlight |

### CODE REVIEW

File: tests/end_to_end/test_brute_force/test_brute_force_rdp/test_brute_force_rdp.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 @pytest.mark.parametrize('metadata', configuration_metadata, ids=cases_ids)
 def test_brute_force(configure_environment, metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```
Comment (juliamagan) [inline tests/end_to_end/test_brute_force/test_brute_force_rdp/test_brute_force_rdp.py:27]:
```suggestion
def test_brute_force_rdp(configure_environment, metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```

File: tests/end_to_end/test_brute_force/test_brute_force_rdp/data/test_cases/cases_brute_force.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  description: Check if the alert is generated when executing a brute force attack via RDP.
+  configuration_parameters: null
+  metadata:
+    rule.id: 60204
+    rule.level: 10
+    rule.description: Multiple Windows logon failures.
+    extra:
+      mitre_technique: Brute Force
+    extra_vars:
+      agent_os: Windows
```
Comment (juliamagan) [inline tests/end_to_end/test_brute_force/test_brute_force_rdp/data/test_cases/cases_brute_force.yaml:11]: We should no longer need this. Also, please rename the file.

File: tests/end_to_end/test_brute_force/test_brute_force_ssh/test_brute_force_ssh.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+events_playbooks = ['generate_events.yaml']
+teardown_playbooks = None
+
+# Configuration
+configurations, configuration_metadata, cases_ids = config.get_test_cases_data(test_cases_file_path)
+
+
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+@pytest.mark.parametrize('metadata', configuration_metadata, ids=cases_ids)
+def test_brute_force(metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```
Comment (juliamagan) [inline tests/end_to_end/test_brute_force/test_brute_force_ssh/test_brute_force_ssh.py:26]:
```suggestion
def test_brute_force_ssh(metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```

File: tests/end_to_end/test_brute_force/test_brute_force_ssh/data/test_cases/cases_brute_force.yaml [hunk 4]
```diff
@@ -9,15 +9,3 @@
       mitre_technique: Brute Force
     extra_vars:
       agent_os: Linux
```
Comment (juliamagan) [inline tests/end_to_end/test_brute_force/test_brute_force_ssh/data/test_cases/cases_brute_force.yaml:11]: Please, rename the file.

File: tests/end_to_end/test_brute_force/test_brute_force_ssh/test_brute_force_ssh.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+from tempfile import gettempdir
+
+from wazuh_testing.tools import configuration as config
+from wazuh_testing import end_to_end as e2e
+from wazuh_testing import event_monitor as evm
+
+# Test cases data
+alerts_json = os.path.join(gettempdir(), 'alerts.json')
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+test_cases_file_path = os.path.join(test_data_path, 'test_cases', 'cases_brute_force.yaml')
```
Comment (juliamagan) [inline tests/end_to_end/test_brute_force/test_brute_force_ssh/test_brute_force_ssh.py:14]:
```suggestion
test_cases_file_path = os.path.join(test_data_path, 'test_cases', 'cases_brute_force_ssh.yaml')
```

File: tests/end_to_end/test_fim/test_fim_linux/data/playbooks/configuration.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+- name: Test case configuration
+  hosts: agents
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_linux/data/playbooks/configuration.yaml:3]:
```suggestion
  hosts: wazuh-agent
```

File: tests/end_to_end/test_fim/test_fim_linux/data/playbooks/configuration.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  tasks:
+
+    - name: Create directory to monitor (Linux)
+      become: True
+      file:
+        path: /tmp/test_demo_fim
+        state: directory
+      when: ansible_facts['system'] == "Linux"
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_linux/data/playbooks/configuration.yaml:11]: We will only run it on the Linux agent, so we will not need to check this.

File: tests/end_to_end/test_fim/test_fim_linux/data/test_cases/cases_fim.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
+- name: create_file_linux
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_linux/data/test_cases/cases_fim.yaml:1]: Rename the file, please.

File: tests/end_to_end/test_fim/test_fim_linux/test_fim_linux.py [hunk 9]
```diff
@@ -22,7 +22,7 @@
 @pytest.mark.parametrize('metadata', configuration_metadata, ids=cases_ids)
 def test_fim(configure_environment, metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_linux/test_fim_linux.py:23]:
```suggestion
def test_fim_linux(configure_environment, metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```

File: tests/end_to_end/test_fim/test_fim_windows/data/playbooks/configuration.yaml [hunk 10]
```diff
@@ -3,30 +3,6 @@
   hosts: agents
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_windows/data/playbooks/configuration.yaml:3]:
```suggestion
  hosts: wazuh-windows
```

File: tests/end_to_end/test_fim/test_fim_windows/data/test_cases/cases_fim.yaml [hunk 11]
```diff
@@ -0,0 +1,41 @@
+- name: create_file_windows
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_windows/data/test_cases/cases_fim.yaml:1]: Rename the file, please.

File: tests/end_to_end/test_fim/test_fim_windows/test_fim_windows.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+configuration_playbooks = ['configuration.yaml']
+events_playbooks = ['generate_events.yaml']
+teardown_playbooks = ['teardown.yaml']
+
+configurations, configuration_metadata, cases_ids = config.get_test_cases_data(test_cases_file_path)
+
+
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+@pytest.mark.parametrize('metadata', configuration_metadata, ids=cases_ids)
+def test_fim(configure_environment, metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```
Comment (juliamagan) [inline tests/end_to_end/test_fim/test_fim_windows/test_fim_windows.py:23]:
```suggestion
def test_fim_windows(configure_environment, metadata, get_dashboard_credentials, generate_events, clean_alerts_index):
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: tests/end_to_end/test_brute_force/linux/data/playbooks/generate_events.yaml [hunk 13]
```diff
@@ -32,23 +32,6 @@
         - "'Permission denied' not in result.stdout"
       when: agent_os == "Linux"
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_brute_force/linux/data/playbooks/generate_events.yaml]: Done in 123ef4c40e7ce10616c8601a498cd48decfe62e7

File: tests/end_to_end/test_brute_force/linux/data/test_cases/cases_brute_force.yaml [hunk 14]
```diff
@@ -9,15 +9,3 @@
       mitre_technique: Brute Force
     extra_vars:
       agent_os: Linux
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_brute_force/linux/data/test_cases/cases_brute_force.yaml]: Done in 123ef4c40e7ce10616c8601a498cd48decfe62e7

File: tests/end_to_end/test_brute_force/test_brute_force_rdp/test_brute_force_rdp.py (hunk 1 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_brute_force/test_brute_force_rdp/test_brute_force_rdp.py]: Done in 123ef4c40e7ce10616c8601a498cd48decfe62e7

File: tests/end_to_end/test_brute_force/test_brute_force_rdp/data/test_cases/cases_brute_force.yaml (hunk 2 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_brute_force/test_brute_force_rdp/data/test_cases/cases_brute_force.yaml]: Done in 123ef4c40e7ce10616c8601a498cd48decfe62e7

File: tests/end_to_end/test_brute_force/test_brute_force_ssh/test_brute_force_ssh.py (hunk 3 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_brute_force/test_brute_force_ssh/test_brute_force_ssh.py]: Done in 123ef4c40e7ce10616c8601a498cd48decfe62e7

File: tests/end_to_end/test_fim/test_fim_linux/data/test_cases/cases_fim.yaml (hunk 8 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_fim/test_fim_linux/data/test_cases/cases_fim.yaml]: Done in 32249dfa1d732e79983d294b7848c8f12e53a5cf

File: tests/end_to_end/test_fim/test_fim_linux/test_fim_linux.py (hunk 9 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_fim/test_fim_linux/test_fim_linux.py]: Done in 32249dfa1d732e79983d294b7848c8f12e53a5cf

File: tests/end_to_end/test_fim/test_fim_windows/data/playbooks/configuration.yaml (hunk 10 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_fim/test_fim_windows/data/playbooks/configuration.yaml]: Done in 32249dfa1d732e79983d294b7848c8f12e53a5cf

File: tests/end_to_end/test_fim/test_fim_windows/data/test_cases/cases_fim.yaml (hunk 11 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_fim/test_fim_windows/data/test_cases/cases_fim.yaml]: Done in 32249dfa1d732e79983d294b7848c8f12e53a5cf

File: tests/end_to_end/test_fim/test_fim_windows/test_fim_windows.py (hunk 12 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_fim/test_fim_windows/test_fim_windows.py]: Done in 32249dfa1d732e79983d294b7848c8f12e53a5cf

**Comment by BelenValdivia** [comment]
Changes for Vulnerability detector tests done in 412dcd8eb09b2cbcaf302661aec2ef65a8137dca

**Comment by BelenValdivia** [comment]
Added folder basic cases:
- To launch these tests, we must indicate the path of the inventory:

```
python -m pytest -vvvs tests/end_to_end/basic_cases/test_fim/ --inventory_path=/home/belen/Repositories/wazuh-qa/tests/end_to_end/general_playbooks/inventory.yml 
```

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**